### PR TITLE
Added GitHub Actions pipeline.

### LINKED ISSUE #237: Add GitHub Actions pipeline.
Replace currently broken Travis CI pipeline with GitHub Actions.

### PR DESCRIPTION
**Pull request by marmelo**
Replaced currently broken Travis CI pipeline with GitHub Actions.
There are some URLs failing that will need to be fixed in future pull requests.

Fixes #237

### CODE REVIEW

**Comment by vpereira01** [comment]
FYI If you want to drop the awesomebot in the future, here's an alternative https://github.com/vpereira01/tech-companies-in-portugal/blob/ghaction-with-node-1/.github/workflows/nodejs.yml

And here's a run output https://github.com/vpereira01/tech-companies-in-portugal/runs/4627733306?check_suite_focus=true

Did this some time ago I think I did it because awesomebot seems too slow for me but it was a long time ago.

**Comment by afonsojramos** [comment]
@marmelo Almost everything is looking to be working fine, there are just 3 URLs that are failing. 
I believe we can remove blockbird.ventures, and remove the careers site from Petapilot.

**Comment by marmelo** [comment]
Hi @vpereira01, I did not knew `remark`.

While `awesome_bot` main objective is to check links, `remark` is markdown processor that also checks for deadlinks.
I did not explore but with `awesome_bot` we can define some acceptable response codes and whitelist links -- which is fundamental for this type of use case.

Regarding time, most times the builds are pretty similar. Although there are some runs that take a lot of time, most likely to be stuck in some URL, which could be mitigated by setting a timeout.

I would keep using `awesome_bot` unless we can replicate our use case in other tool and there is a clear advantage.
What do you/others think?

**Comment by marmelo** [comment]
Hi @carlosfaria94, some time ago you added blockbird.ventures which website is not gone.
Can you please inform us if the company still exists?

Thank you.

**Comment by marmelo** [comment]
Hi @picadoh, can you please inform us if there is a new careers link for Petapilot?

Thank you.

**Comment by marmelo** [comment]
Hi @afonsojramos thank you for merging the latest changes. I agree to remove all the dead entries and links.
Lets just wait a couple days for the users who added those companies to provide more info.

Thanks.

**Comment by carlosfaria94** [comment]
> Hi @carlosfaria94, some time ago you added blockbird.ventures which website is now gone (parked at NameCheap). Can you please inform us if the company still exists?
> 
> Thank you.

Hi @marmelo, blockbird.ventures don't exist anymore

Thank you

**Review by afonsojramos — DISMISSED** [review]

**Review by afonsojramos — DISMISSED** [review]

**Comment by vpereira01** [comment]
> Hi @vpereira01, I did not knew `remark`.
> 
> While `awesome_bot` main objective is to check links, `remark` is markdown processor that also checks for deadlinks. I did not explore but with `awesome_bot` we can define some acceptable response codes and whitelist links -- which is fundamental for this type of use case.
> 
> Regarding time, most times the builds are pretty similar. Although there are some runs that take a lot of time, most likely to be stuck in some URL, which could be mitigated by setting a timeout.
> 
> I would keep using `awesome_bot` unless we can replicate our use case in other tool and there is a clear advantage. What do you/others think?

Seems good to keep awesome_bot.

PS: Sorry for only submitting PR comments now, I didn't realize my comments were "pending" and only visible to me.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: CI
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build:
```
[the remark this answers is not in the document]
Comment (afonsojramos) [reply .github/workflows/main.yml:10]: At first, in my mind, I disagreed, but it does make sense!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: CI
+
+on:
```
[the remark this answers is not in the document]
Comment (afonsojramos) [reply .github/workflows/main.yml:3]: They do if it happens on main, but otherwise, agreed!

[the remark this answers is not in the document]
Comment (marmelo) [reply .github/workflows/main.yml:3]: Agree.

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marmelo) [reply .github/workflows/main.yml:10]: AFAIK public repositories have free unlimited actions.
But I am defining a 5 min limit regardless (which may be tight sometimes).

**Comment by marmelo** [comment]
Changes done @vpereira01 @afonsojramos, please check.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on: pull_request
```
Comment (afonsojramos) [inline .github/workflows/main.yml:3]: What about changing this to:

```
on:
  push:
    branches: [master]
  pull_request:
    branches: [master]
```

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpereira01) [reply .github/workflows/main.yml:6]: Just rechecked, you're right for public repositories there seems to be no limit.

https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

File: .github/workflows/main.yml (hunk 3 above)
Comment (afonsojramos) [reply .github/workflows/main.yml:3]: @marmelo

Comment (afonsojramos) [reply .github/workflows/main.yml:3]: @vpereira01 @pedrorijo91 what do you think? This workflow seems ready to merge with or without this change.

Comment (marmelo) [reply .github/workflows/main.yml]: Just updated this.

**Comment by marmelo** [comment]
Build failed because it took more than 5 minutes, I've just doubled it to 10 minutes.
If we feel in the future that is too much we can review it.

**Review by vpereira01 — APPROVED** [review]
Overall it's an improvement. The pipeline still fails due to network connection issues with some sites but provides a greater feedback than before.

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**